### PR TITLE
Added SearchField-component

### PR DESCRIPTION
**Pull request by rasmuswoelk**
The SearchField component is an universal component that will be used in different places across the application to prevent duplicate code. It's simple for now but can be updated later on with more functionality if needed.

**Changes:**
- Added SearchField component and story under "structures"
- Updated Icon component with id prop. Made it a stateless component.
- Added new props to TextField (ariaLabel) and TextFieldIcon (id)

### CODE REVIEW

File: lib/structures/SearchField/SearchField.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
+/**
+ * SearchFieldUsage
```
Comment (doytch) [inline lib/structures/SearchField/SearchField.js]: Super minor but is this comment supposed to just be `SearchField` for autodocs?

**Comment by doytch** [comment]
@MikeTaylor the Codex Search looked solid in the demo today! One thing: it looks like you're using the `<select>` functionality of `FilterPaneSearch` to do the selection of what the search field is searching (title, ISBN, etc). 

So @rasmuswoelk , I think we'll need that same functionality in this SearchField component as well if we want to move it off of the header as per [STCOM-164 #1](https://issues.folio.org/browse/STCOM-164)

**Comment by rasmuswoelk** [comment]
@doytch I haven't seen the demo yet - I will check it out and get back to this 👍

**Comment by rasmuswoelk** [comment]
@doytch I have updated the component to support searchable indexes. Can you check it out and let me know if it supports what you need?

**Comment by doytch** [comment]
Slick! Looks good man!

**Pull request merged**